### ISSUE TITLE
Index site 1337x download links dead

### ISSUE THREAD

**Issue by binhex** [Jackett/Jackett#3407]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue. Duplicating issues makes it more difficult for everyone to follow. Your issue may have already been solved in the past as well.

Provide a description of the feature request or bug, the more details the better.
If you are experiencing an issue with a tracker, a **full enhanced log must be included**. Instructions for obtaining logs are here: https://github.com/Jackett/Jackett#troubleshooting

**Jackett version**:  0.9.6.0
**Mono version** (if not using Windows):   5.10.0

-------------------------------
There are currently two issues with index site 1337x in jackett, first issue and most importantly is that all download links timeout and do not successfully download a torrent (this is with either magnet or itorrents.org as download link type). The second issue is that even if you select a 'download link' type of magnet the results returned are for torrent downloads (which dont work, see first issue). 

In short 1337x is currently tests fine, does return results but you cannot download anything (torrent or working magnet link).

I see you normally want logs for index issues and im willing to post them but i think its such an obvious issue that im pretty sure you wll be able to replicate.

**Comment by binhex** [comment]
Tested this again this morning and i can now download successfully when download link type of itorrent is selected, perhaps problems with itorrent.org?, in any case issue #2 persists here, perhaps worth investigation, see log file for replication of issue.
[1337_log.txt](https://github.com/Jackett/Jackett/files/2202027/1337_log.txt)

**Comment by kaso17** [comment]
please see #3475

**Comment by binhex** [comment]
Please re-open this issue, even if you select download type of 'magnet" as suggested in the resolution the search results returned do not show magnet links (shows torrent downloads, which of course fail) and thus currently 1337x is unusable

**Comment by kaso17** [comment]
@binhex this is the expected behavior for the jackett manual search. It can't show a direct magnet link because it's not available on the search results page. If you click on the download link it should redirect you to the magnet link.

**Comment by binhex** [comment]
@kaso17 it does show magnet links for other manual searches, do  a manual search against index site The Pirate Bay, the links returned on the right under the download column are magnet links, not links from jackett to magnet links.

if i click on the download link for any of the returned results for 1337X then it does not redirect at all.

**Comment by kaso17** [comment]
For other indexers direct magnet links are shown (if they're available on the corresponding search results page).
If you're using chrome you can see the redirect in the developer console. On firefox you can't see the redirect at all. But assuming the magnet: protocol is registered with a torrent client it should work perfectly fine.

**Comment by binhex** [comment]
i see no redirection happening with chrome, in fact i don't even see a timeout, it simply does nothing when i click on the download link, ive disabled ad blocker, disabled scriptsafe and still nothing, if i copy and paste the link to my torrent client (rutorrent/rtorrent) then it fails to load, am i doing something wrong here, does this just work for you?.

**Comment by kaso17** [comment]
![magnetdl](https://user-images.githubusercontent.com/11887264/43533707-bb6db762-95b5-11e8-8cb0-9ec4ad5f8c7c.png)
That's how it should look in the chrome dev console. It just works for me.

**Comment by binhex** [comment]
ok so i see the redirection in dev console and i see the magnet link, however neither deluge or rutorrent can follow the direction and load the magnet link, can you get this to work on any known torrent clients?.

**Comment by kaso17** [comment]
For me it works with e.g. utorrent. I assume you're using remote deluge/rutorrent installations. Which extension/application are you using to link torrent/magnet files to these?

**Comment by binhex** [comment]
yes im using a web interface to deluge/rutorrent, i was simply trying to copy and paste the link shown in the downloads column for jackett, unfortunately it looks like deluge/rutorrent do not follow redirects and thus the magnet link is not resolved and loaded.

i can see where you are coming from with this, where utorrent will load the magnet link detecting it from the redirection the browser will make, as utorrent will be associated with magnet lnks, whereas a web based client does not pick up magnet/torrent files (im aware of browser plugins that CAN do this but they generally dont work too well) you have to manually add them in.

Im assuming changing this from redirection to magnet links to the alternative method that TPB uses where you get the direct magnet link is tricky/impossible for this index site?.

p.s. thanks for replying btw, i realise this is probably sucking up your time, so i appreciate it.

**Comment by kaso17** [comment]
To get the direct magnet link for these indexers we would have to grab the details page of each release during the search. That would make the search very slow and probably would get your IP banned for to many request very fast.

I suggest you open issues with deluge/rutorrent to add support for "redirect to magnet links". Shouldn't be to hard to implement that.

**Comment by binhex** [comment]
ok thats cool, i get that's not your problem, thanks again for working through the issue, at least i now understand a little more how jackett handles magnet links, hopefully this thread will enlighten someone else in the future :-).
